### ISSUE TITLE
How to manually set initial points?

### ISSUE THREAD

**Issue by TianningGao** [automl/SMAC3#1048]
#### Description
I want to manually set the initial points (or initial configurations) but get some error. For example, I have my initial points in numpy array 'X' and their evaluations in 'Y'. I tried adding them with runhistory.add() and python reports AssertionError while running "optimize()".
I saw someone mentioned passing in initial points through initial_design (e.g. DefaultInitialDesign). How can I correctly pass in both the initial points and evaluations stored in numpy array?

#### Steps/Code to Reproduce
```
import numpy as np
import os
from typing import List, Union, Callable
from ConfigSpace import Configuration, ConfigurationSpace
from ConfigSpace import Categorical, Float, Integer
from smac import HyperparameterOptimizationFacade, Scenario
from smac.initial_design import SobolInitialDesign

def my_func(config: Configuration, seed=0):
    x = config.get_array()
    x[0] *= 1e-2
    x[1] *= 1e-2

    part1 = np.square( np.sin( np.square(x[0]) - np.square(x[1]) ) )

    part2 = np.square(x[:2]).sum()

    part3 = 0
    for xi in x[2:5]:
        for xj in x[2:5]:
            part3 += xi*xj + (1-xi)*(1-xj)

    return 0.5 + (part1 - 0.5) / np.square(1 + 0.01* part2) + part3 / 7

cs = ConfigurationSpace()
x0 = Integer('x0', (-150,150)); cs.add_hyperparameter(x0)
x1 = Integer('x1', (-150,150)); cs.add_hyperparameter(x1)
x2 = Categorical('x2', [0,1]); cs.add_hyperparameter(x2)
x3 = Categorical('x3', [0,1]); cs.add_hyperparameter(x3)
x4 = Categorical('x4', [0,1]); cs.add_hyperparameter(x4)
sce = Scenario(
    configspace=cs,
    deterministic=True,
    n_trials = 12
)

X = np.hstack((np.random.randint(-150,151,size=(8,2)), np.random.randint(0,2,size=(8,3)) ))
X_cfg = []
Y = []
for i in range(8):
    x_cfg = Configuration(cs,vector=X[i])
    X_cfg.append( x_cfg )
    y = my_func( x_cfg )
    Y.append(y)

opt = HyperparameterOptimizationFacade(
    scenario= sce,
    target_function= my_func,
    # initial_design= init_design,
    overwrite= True
)

for i in range(8):
    opt.runhistory.add(X_cfg[i],Y[i], seed=0)

incumbent = opt.optimize()
print(incumbent.get_array(), opt.validate(incumbent))
```

#### Expected Results
Incumbent and its evaluation

#### Actual Results
```
idwwwoqq808@RyzenTR:~/workspace/optimizer/EnsOpt/tmp$ py test.py
[INFO][abstract_initial_design.py:147] Using 2 initial design configurations and 8 additional configurations.
[INFO][abstract_intensifier.py:305] Using only one seed for deterministic scenario.
[INFO][abstract_intensifier.py:515] Added config 058091 as new incumbent because there are no incumbents yet.
Traceback (most recent call last):
  File "/home/idwwwoqq808/workspace/optimizer/EnsOpt/tmp/test.py", line 66, in <module>
    incumbent = opt.optimize()
  File "/home/idwwwoqq808/.local/lib/python3.10/site-packages/smac/facade/abstract_facade.py", line 314, in optimize
    incumbents = self._optimizer.optimize(data_to_scatter=data_to_scatter)
  File "/home/idwwwoqq808/.local/lib/python3.10/site-packages/smac/main/smbo.py", line 300, in optimize
    trial_info = self.ask()
  File "/home/idwwwoqq808/.local/lib/python3.10/site-packages/smac/main/smbo.py", line 153, in ask
    trial_info = next(self._trial_generator)
  File "/home/idwwwoqq808/.local/lib/python3.10/site-packages/smac/intensifier/intensifier.py", line 226, in __iter__
    config = next(self.config_generator)
  File "/home/idwwwoqq808/.local/lib/python3.10/site-packages/smac/main/config_selector.py", line 222, in __iter__
    for config in challengers:
  File "/home/idwwwoqq808/.local/lib/python3.10/site-packages/smac/acquisition/maximizer/helpers.py", line 62, in __next__
    self._challengers = self._challengers_callback()
  File "/home/idwwwoqq808/.local/lib/python3.10/site-packages/smac/acquisition/maximizer/abstract_acqusition_maximizer.py", line 108, in next_configs_by_acquisition_value
    return [t[1] for t in self._maximize(previous_configs, n_points)]
  File "/home/idwwwoqq808/.local/lib/python3.10/site-packages/smac/acquisition/maximizer/local_and_random_search.py", line 115, in _maximize
    next_configs_by_local_search = self._local_search._maximize(
  File "/home/idwwwoqq808/.local/lib/python3.10/site-packages/smac/acquisition/maximizer/local_search.py", line 118, in _maximize
    configs_acq = self._search(init_points)
  File "/home/idwwwoqq808/.local/lib/python3.10/site-packages/smac/acquisition/maximizer/local_search.py", line 349, in _search
    n = next(neighborhood_iterator)
  File "/home/idwwwoqq808/.local/lib/python3.10/site-packages/ConfigSpace/util.py", line 199, in get_one_exchange_neighbourhood
    neighbors = hp.get_neighbors(
  File "ConfigSpace/hyperparameters/uniform_integer.pyx", line 221, in ConfigSpace.hyperparameters.uniform_integer.UniformIntegerHyperparameter.get_neighbors
AssertionError: For get neighbors of UniformIntegerHyperparameter, the value if assumed to be in the unit-hypercube [0, 1]. If this was not the behaviour assumed, please raise a ticket on github.
```

#### Versions

SMAC3: 2.0.1
ConfigSpace: 0.7.1
OS: Ubuntu 22.04
Python: 3.10.6

**Comment by TianningGao** [comment]
I tried passing initial points with SobolInitialDesign and I still got AssertionError. This code works fine if I delete line "additional_configs = X_cfg" but that would not allow me to pass in my initial points.
**Code**
```
def my_func(config: Configuration, seed=0):
    x = config.get_array()
    x[0] *= 1e-2
    x[1] *= 1e-2
    part1 = np.square( np.sin( np.square(x[0]) - np.square(x[1]) ) )
    part2 = np.square(x[:2]).sum()
    part3 = 0
    for xi in x[2:5]:
        for xj in x[2:5]:
            part3 += xi*xj + (1-xi)*(1-xj)

    return 0.5 + (part1 - 0.5) / np.square(1 + 0.01* part2) + part3 / 7

cs = ConfigurationSpace()
x0 = Integer('x0', (-150,150)); cs.add_hyperparameter(x0)
x1 = Integer('x1', (-150,150)); cs.add_hyperparameter(x1)
x2 = Categorical('x2', [0,1]); cs.add_hyperparameter(x2)
x3 = Categorical('x3', [0,1]); cs.add_hyperparameter(x3)
x4 = Categorical('x4', [0,1]); cs.add_hyperparameter(x4)
sce = Scenario(
    configspace=cs,
    deterministic=True,
    n_trials = 12
)

X = np.hstack((np.random.randint(-150,151,size=(8,2)), np.random.randint(0,2,size=(8,3)) ))

X_cfg = []
Y = []
for i in range(8):
    x_cfg = Configuration(cs,vector=X[i])
    X_cfg.append( x_cfg )

init_design = SobolInitialDesign(
    scenario = sce,
    n_configs = 8,
    additional_configs = X_cfg
)

opt = HyperparameterOptimizationFacade(
    scenario= sce,
    target_function= my_func,
    initial_design= init_design,
    overwrite= True
)

for i in range(8):
    opt.runhistory.add(X_cfg[i],Y[i], seed=0)

incumbent = opt.optimize()
print(incumbent.get_array(), opt.validate(incumbent))
```
**Terminal Output**
```
[INFO][abstract_initial_design.py:147] Using 3 initial design configurations and 8 additional configurations.
[INFO][abstract_intensifier.py:515] Added config 78fc64 as new incumbent because there are no incumbents yet.
[INFO][abstract_intensifier.py:590] Added config 472843 and rejected config 78fc64 as incumbent because it is not better than the incumbents on 1 instances:
[INFO][configspace.py:175] --- x0: -150 -> -27
[INFO][configspace.py:175] --- x1: -150 -> -48
[INFO][configspace.py:175] --- x3: 0 -> 1
[INFO][configspace.py:175] --- x4: 1 -> 0
[INFO][intensifier.py:126] Added config 78fc64 from runhistory to the intensifier queue.
[INFO][intensifier.py:126] Added config a83354 from runhistory to the intensifier queue.
[INFO][intensifier.py:126] Added config 472843 from runhistory to the intensifier queue.
[INFO][intensifier.py:126] Added config b7f800 from runhistory to the intensifier queue.
[INFO][intensifier.py:126] Added config 14f5b7 from runhistory to the intensifier queue.
[INFO][intensifier.py:126] Added config 53d413 from runhistory to the intensifier queue.
[INFO][intensifier.py:126] Added config dd3112 from runhistory to the intensifier queue.
[INFO][intensifier.py:126] Added config 05c0cc from runhistory to the intensifier queue.
[INFO][abstract_intensifier.py:287] Added existing seed 0 from runhistory to the intensifier.
[INFO][abstract_intensifier.py:305] Using only one seed for deterministic scenario.
[INFO][abstract_intensifier.py:590] Added config 058091 and rejected config 472843 as incumbent because it is not better than the incumbents on 1 instances:
[INFO][configspace.py:175] --- x0: -27 -> -70
[INFO][configspace.py:175] --- x1: -48 -> 36
[INFO][configspace.py:175] --- x3: 1 -> 0
[INFO][configspace.py:175] --- x4: 0 -> 1
Traceback (most recent call last):
  File "/home/idwwwoqq808/workspace/optimizer/EnsOpt/tmp/test.py", line 66, in <module>
    incumbent = opt.optimize()
  File "/home/idwwwoqq808/.local/lib/python3.10/site-packages/smac/facade/abstract_facade.py", line 314, in optimize
    incumbents = self._optimizer.optimize(data_to_scatter=data_to_scatter)
  File "/home/idwwwoqq808/.local/lib/python3.10/site-packages/smac/main/smbo.py", line 300, in optimize
    trial_info = self.ask()
  File "/home/idwwwoqq808/.local/lib/python3.10/site-packages/smac/main/smbo.py", line 153, in ask
    trial_info = next(self._trial_generator)
  File "/home/idwwwoqq808/.local/lib/python3.10/site-packages/smac/intensifier/intensifier.py", line 226, in __iter__
    config = next(self.config_generator)
  File "/home/idwwwoqq808/.local/lib/python3.10/site-packages/smac/main/config_selector.py", line 222, in __iter__
    for config in challengers:
  File "/home/idwwwoqq808/.local/lib/python3.10/site-packages/smac/acquisition/maximizer/helpers.py", line 62, in __next__
    self._challengers = self._challengers_callback()
  File "/home/idwwwoqq808/.local/lib/python3.10/site-packages/smac/acquisition/maximizer/abstract_acqusition_maximizer.py", line 108, in next_configs_by_acquisition_value
    return [t[1] for t in self._maximize(previous_configs, n_points)]
  File "/home/idwwwoqq808/.local/lib/python3.10/site-packages/smac/acquisition/maximizer/local_and_random_search.py", line 115, in _maximize
    next_configs_by_local_search = self._local_search._maximize(
  File "/home/idwwwoqq808/.local/lib/python3.10/site-packages/smac/acquisition/maximizer/local_search.py", line 118, in _maximize
    configs_acq = self._search(init_points)
  File "/home/idwwwoqq808/.local/lib/python3.10/site-packages/smac/acquisition/maximizer/local_search.py", line 349, in _search
    n = next(neighborhood_iterator)
  File "/home/idwwwoqq808/.local/lib/python3.10/site-packages/ConfigSpace/util.py", line 199, in get_one_exchange_neighbourhood
    neighbors = hp.get_neighbors(
  File "ConfigSpace/hyperparameters/uniform_integer.pyx", line 221, in ConfigSpace.hyperparameters.uniform_integer.UniformIntegerHyperparameter.get_neighbors
AssertionError: For get neighbors of UniformIntegerHyperparameter, the value if assumed to be in the unit-hypercube [0, 1]. If this was not the behaviour assumed, please raise a ticket on github.
```

**Comment by eddiebergman** [comment]
Hi @TianningGao,

As far as I'm aware, you can use `optimizer.tell()` with your initial points to _tell_ SMAC about your previous evaluations before beginning.

Initial Design is more about telling SMAC where to begin looking while using `.tell()` is more about telling SMAC about some prior evaluations you have.

Here is a snippet of code from another project where I did so:

```python
@dataclass
class WarmstartConfig:
    config: Configuration
    seed: int
    cost: float
    duration: float
    budget: float | None = None

    def as_trial(self) -> tuple[TrialInfo, TrialValue]:
        # Since we're not using Multi-fidelity, budget=self.budget,
        trial_info = TrialInfo(config=self.config, instance=None, seed=self.seed)
        trial_value = TrialValue(time=self.duration, cost=self.cost)
        return trial_info, trial_value
        
# Later on
optimizer = HyperparameterOptimizationFacade(
    target_function=target_function,
    scenario=scenario,
    overwrite=args.overwrite,
    logging_level=args.log_level,
    initial_design=init_design,
)

for config in meta_configs:
    trial_info, trial_value = config.as_trial()
    optimizer.tell(trial_info, trial_value)
```

I hope this helps :)

**Comment by TianningGao** [comment]
Hello @eddiebergman 

I tried your method and I still got the same error.
**Code**
```
def my_func(config: Configuration, seed=0):
    x = config.get_array()
    x[0] *= 1e-2
    x[1] *= 1e-2

    part1 = np.square( np.sin( np.square(x[0]) - np.square(x[1]) ) )

    part2 = np.square(x[:2]).sum()

    part3 = 0
    for xi in x[2:5]:
        for xj in x[2:5]:
            part3 += xi*xj + (1-xi)*(1-xj)

    return 0.5 + (part1 - 0.5) / np.square(1 + 0.01* part2) + part3 / 7

cs = ConfigurationSpace()
x0 = Integer('x0', (-150,150)); cs.add_hyperparameter(x0)
x1 = Integer('x1', (-150,150)); cs.add_hyperparameter(x1)
x2 = Categorical('x2', [0,1]); cs.add_hyperparameter(x2)
x3 = Categorical('x3', [0,1]); cs.add_hyperparameter(x3)
x4 = Categorical('x4', [0,1]); cs.add_hyperparameter(x4)

sce = Scenario(
    configspace=cs,
    deterministic=True,
    n_trials = 20,
    # n_workers = 1
)
init_design = SobolInitialDesign(
    scenario = sce,
    n_configs = 8
)

opt = HyperparameterOptimizationFacade(
    scenario= sce,
    target_function= my_func,
    initial_design= init_design,
    overwrite= True
)

X = np.hstack((np.random.randint(-150,150,size=(8,2)), np.random.randint(0,2,size=(8,3)) ))
for i in range(8):
    x_cfg = Configuration(cs,vector=X[i])
    y = my_func( x_cfg )
    trial_info = TrialInfo(config= x_cfg, instance= None, seed= 0)
    trial_value = TrialValue(time = 1, cost= y)
    opt.tell(info = trial_info, value= trial_value)

incumbent = opt.optimize()
print(incumbent.get_array(), opt.validate(incumbent))
```
**Terminal Output**
```
[INFO][abstract_initial_design.py:147] Using 5 initial design configurations and 0 additional configurations.
[INFO][abstract_intensifier.py:515] Added config f82587 as new incumbent because there are no incumbents yet.
[INFO][abstract_intensifier.py:590] Added config d79544 and rejected config f82587 as incumbent because it is not better than the incumbents on 1 instances:
[INFO][configspace.py:175] --- x0: -150 -> -141
[INFO][configspace.py:175] --- x1: 150 -> -150
[INFO][configspace.py:175] --- x2: 0 -> 1
[INFO][intensifier.py:126] Added config f82587 from runhistory to the intensifier queue.
[INFO][intensifier.py:126] Added config d79544 from runhistory to the intensifier queue.
[INFO][intensifier.py:126] Added config b0b826 from runhistory to the intensifier queue.
[INFO][intensifier.py:126] Added config 33823f from runhistory to the intensifier queue.
[INFO][intensifier.py:126] Added config 734729 from runhistory to the intensifier queue.
[INFO][intensifier.py:126] Added config 38a2d5 from runhistory to the intensifier queue.
[INFO][intensifier.py:126] Added config f443ce from runhistory to the intensifier queue.
[INFO][intensifier.py:126] Added config d4f7fb from runhistory to the intensifier queue.
[INFO][abstract_intensifier.py:287] Added existing seed 0 from runhistory to the intensifier.
[INFO][abstract_intensifier.py:305] Using only one seed for deterministic scenario.
[INFO][abstract_intensifier.py:590] Added config 058091 and rejected config d79544 as incumbent because it is not better than the incumbents on 1 instances:
[INFO][configspace.py:175] --- x0: -141 -> -70
[INFO][configspace.py:175] --- x1: -150 -> 36
[INFO][configspace.py:175] --- x2: 1 -> 0
[INFO][configspace.py:175] --- x4: 0 -> 1
Traceback (most recent call last):
  File "/home/idwwwoqq808/workspace/optimizer/EnsOpt/tmp/test.py", line 59, in <module>
    incumbent = opt.optimize()
  File "/home/idwwwoqq808/.local/lib/python3.10/site-packages/smac/facade/abstract_facade.py", line 314, in optimize
    incumbents = self._optimizer.optimize(data_to_scatter=data_to_scatter)
  File "/home/idwwwoqq808/.local/lib/python3.10/site-packages/smac/main/smbo.py", line 300, in optimize
    trial_info = self.ask()
  File "/home/idwwwoqq808/.local/lib/python3.10/site-packages/smac/main/smbo.py", line 153, in ask
    trial_info = next(self._trial_generator)
  File "/home/idwwwoqq808/.local/lib/python3.10/site-packages/smac/intensifier/intensifier.py", line 226, in __iter__
    config = next(self.config_generator)
  File "/home/idwwwoqq808/.local/lib/python3.10/site-packages/smac/main/config_selector.py", line 222, in __iter__
    for config in challengers:
  File "/home/idwwwoqq808/.local/lib/python3.10/site-packages/smac/acquisition/maximizer/helpers.py", line 62, in __next__
    self._challengers = self._challengers_callback()
  File "/home/idwwwoqq808/.local/lib/python3.10/site-packages/smac/acquisition/maximizer/abstract_acqusition_maximizer.py", line 108, in next_configs_by_acquisition_value
    return [t[1] for t in self._maximize(previous_configs, n_points)]
  File "/home/idwwwoqq808/.local/lib/python3.10/site-packages/smac/acquisition/maximizer/local_and_random_search.py", line 115, in _maximize
    next_configs_by_local_search = self._local_search._maximize(
  File "/home/idwwwoqq808/.local/lib/python3.10/site-packages/smac/acquisition/maximizer/local_search.py", line 118, in _maximize
    configs_acq = self._search(init_points)
  File "/home/idwwwoqq808/.local/lib/python3.10/site-packages/smac/acquisition/maximizer/local_search.py", line 349, in _search
    n = next(neighborhood_iterator)
  File "/home/idwwwoqq808/.local/lib/python3.10/site-packages/ConfigSpace/util.py", line 199, in get_one_exchange_neighbourhood
    neighbors = hp.get_neighbors(
  File "ConfigSpace/hyperparameters/uniform_integer.pyx", line 221, in ConfigSpace.hyperparameters.uniform_integer.UniformIntegerHyperparameter.get_neighbors
AssertionError: For get neighbors of UniformIntegerHyperparameter, the value if assumed to be in the unit-hypercube [0, 1]. If this was not the behaviour assumed, please raise a ticket on github.
```

**Comment by dengdifan** [comment]
Hi,
thanks for the information. Indeed, all the integral hyperparameters are stored as vectors that range from [0,1],
 therefore, you need to initialize the configuration in the following way:
```
X = np.hstack((np.random.randint(-0, 1,size=(8,2)), np.random.randint(0,2,size=(8,3)) ))
for i in range(8):
    x_cfg = Configuration(cs,vector=X[i])
```
or 
```
X = np.hstack((np.random.randint(-150, 150,size=(8,2)), np.random.randint(0,2,size=(8,3)) ))
for i in range(8):
    values = {f'x{j}': X[i, j] for j in range(5)}
    x_cfg = Configuration(cs,values=values)
```
Correspondingly, `myfun` needs to be adapted by:
```
def my_func(config: Configuration, seed=0):
   x = list(config.get_dictionary().values())
    ....
``` 
@eddiebergman seems that `ConfigSpace` does not check if the `initial_vector` value is valid for the corresponding hyperparameter, maybe we could check that during initialization

**Comment by alexandertornede** [comment]
@TianningGao Is your problem fixed with @dengdifan's comment above?

**Comment by TianningGao** [comment]
Hello,
Sorry for the late reply. I tried second method by @dengdifan and it worked without a problem. Thanks for all your help!

Here's the full code:
```
import numpy as np
import os, copy
from typing import List, Union, Callable
from ConfigSpace import Configuration, ConfigurationSpace
from ConfigSpace import Categorical, Float, Integer
from smac import HyperparameterOptimizationFacade, Scenario
from smac.initial_design import SobolInitialDesign
from smac.runhistory import TrialInfo, TrialValue

def my_func(config: Configuration, seed=0):
    x = config.get_array()
    x[0] *= 1e-2
    x[1] *= 1e-2

    part1 = np.square( np.sin( np.square(x[0]) - np.square(x[1]) ) )

    part2 = np.square(x[:2]).sum()

    part3 = 0
    for xi in x[2:5]:
        for xj in x[2:5]:
            part3 += xi*xj + (1-xi)*(1-xj)

    return 0.5 + (part1 - 0.5) / np.square(1 + 0.01* part2) + part3 / 7

cs = ConfigurationSpace()
x0 = Integer('x0', (-150,150)); cs.add_hyperparameter(x0)
x1 = Integer('x1', (-150,150)); cs.add_hyperparameter(x1)
x2 = Categorical('x2', [0,1]); cs.add_hyperparameter(x2)
x3 = Categorical('x3', [0,1]); cs.add_hyperparameter(x3)
x4 = Categorical('x4', [0,1]); cs.add_hyperparameter(x4)

sce = Scenario(
    configspace=cs,
    deterministic=True,
    n_trials = 20
)

opt = HyperparameterOptimizationFacade(
    scenario= sce,
    target_function= my_func,
    overwrite= True
)

X = np.hstack((np.random.randint(-150,150,size=(8,2)), np.random.randint(0,2,size=(8,3)) ))
for i in range(8):
    x_dict = {f'x{j}': X[i, j].item() for j in range(5)}
    x_cfg = Configuration(cs,values=x_dict)
    y = my_func( x_cfg )
    trial_info = TrialInfo(config= x_cfg, instance= None, seed= 0)
    trial_value = TrialValue(time = 1, cost= y)
    opt.tell(info = trial_info, value= trial_value)

incumbent = opt.optimize()
print(incumbent.get_array(), opt.validate(incumbent)); exit()
```
Here's the terminal output:
```
[INFO][abstract_initial_design.py:95] Reducing the number of initial configurations from 50 to 5 (max_ratio == 0.25).
[INFO][abstract_initial_design.py:147] Using 5 initial design configurations and 0 additional configurations.
[INFO][abstract_intensifier.py:515] Added config bc750d as new incumbent because there are no incumbents yet.
[INFO][abstract_intensifier.py:590] Added config ce32be and rejected config bc750d as incumbent because it is not better than the incumbents on 1 instances:
[INFO][configspace.py:175] --- x0: -35 -> -6
[INFO][configspace.py:175] --- x1: -104 -> -88
[INFO][configspace.py:175] --- x4: 0 -> 1
[INFO][abstract_intensifier.py:590] Added config 56ab38 and rejected config ce32be as incumbent because it is not better than the incumbents on 1 instances:
[INFO][configspace.py:175] --- x0: -6 -> -11
[INFO][configspace.py:175] --- x1: -88 -> -126
[INFO][configspace.py:175] --- x3: 0 -> 1
[INFO][abstract_intensifier.py:590] Added config 78d1f3 and rejected config 56ab38 as incumbent because it is not better than the incumbents on 1 instances:
[INFO][configspace.py:175] --- x0: -11 -> -52
[INFO][configspace.py:175] --- x1: -126 -> -66
[INFO][configspace.py:175] --- x4: 1 -> 0
[INFO][intensifier.py:126] Added config bc750d from runhistory to the intensifier queue.
[INFO][intensifier.py:126] Added config ce32be from runhistory to the intensifier queue.
[INFO][intensifier.py:126] Added config 56ab38 from runhistory to the intensifier queue.
[INFO][intensifier.py:126] Added config 78d1f3 from runhistory to the intensifier queue.
[INFO][intensifier.py:126] Added config 1f7f11 from runhistory to the intensifier queue.
[INFO][intensifier.py:126] Added config 025268 from runhistory to the intensifier queue.
[INFO][intensifier.py:126] Added config c953d0 from runhistory to the intensifier queue.
[INFO][intensifier.py:126] Added config ad57be from runhistory to the intensifier queue.
[INFO][abstract_intensifier.py:287] Added existing seed 0 from runhistory to the intensifier.
[INFO][abstract_intensifier.py:305] Using only one seed for deterministic scenario.
[INFO][abstract_intensifier.py:590] Added config 1883df and rejected config 78d1f3 as incumbent because it is not better than the incumbents on 1 instances:
[INFO][configspace.py:175] --- x0: -52 -> -150
[INFO][configspace.py:175] --- x1: -66 -> -150
[INFO][smbo.py:327] Configuration budget is exhausted:
[INFO][smbo.py:328] --- Remaining wallclock time: inf
[INFO][smbo.py:329] --- Remaining cpu time: inf
[INFO][smbo.py:330] --- Remaining trials: 0
[INFO][abstract_intensifier.py:305] Using only one seed for deterministic scenario.
[0.00327243 0.00280731 0.         1.         0.        ] 0.7142857143043041
```

**Comment by alexandertornede** [comment]
@TianningGao Thank you very much for your feedback! I will close the issue then. Feel free to re-open, if something else regarding the same problem pops up.

**Comment by eismont21** [comment]
I encountered another problem while informing SMAC about my previous evaluations before starting. When providing too many data points (e.g., 100), the optimization algorithm seems to select the best one among them and stops, disregarding the specified n_trials of 1000 in the scenario.
```
[ERROR][intensifier.py:134] Intensifier could not find any new trials.
[INFO][smbo.py:319] Finished 100 trials.
[INFO][smbo.py:332] Shutting down because the stop flag was set.
```

It seems that there might be an issue with the 'retries' parameter of Intensifier. By default, 'retries' is set to 16, which is lower than the number of data points I provided. However, when I increased it to 200, the optimization process started working as expected. While this change seems to resolve the problem, I'm not entirely certain if it's the correct solution. I would appreciate any input on whether adjusting this parameter to a higher value like 200 is a valid approach or if there might be a more appropriate way to address the issue.
